### PR TITLE
gh-93678: add _testinternalcapi.optimize_cfg() and test utils for compiler optimization unit tests

### PR DESCRIPTION
**Pull request by iritkatriel**
* Issue: gh-93678

### CODE REVIEW

File: Python/compile.c [hunk 1]
```diff
@@ -9594,6 +9605,157 @@ duplicate_exits_without_lineno(cfg_builder *g)
...
+        }
+        if (PyList_Append(instructions, lbl) != 0) {
+            Py_DECREF(lbl);
+            goto error;
+        }
+        Py_DECREF(lbl);
+        for (int i = 0; i < b->b_iused; i++) {
+            struct instr *instr = &b->b_instr[i];
+            struct location loc = instr->i_loc;
+            long long arg = instr->i_oparg;
```
Comment (markshannon) [inline Python/compile.c]: Why 64bit, not 32bit?

Comment (iritkatriel) [reply Python/compile.c]: Two lines down, ``instr->i_target`` is a pointer, and I'm passing its address back to python as the label of the block. (It get normalised by the test harness).

Comment (markshannon) [reply Python/compile.c]: If it is an address, it should be `uintptr_t`, and add a comment that it just an ID.

Comment (iritkatriel) [reply Python/compile.c]: With uintptr_t the test fail on the Windows x86 buildbot.

Comment (iritkatriel) [reply Python/compile.c]: I changed to (small) calculated block IDs, so we don't have the issue anymore.

**Comment by iritkatriel** [comment]
The freebsd test failure seems unrelated and affects other PRs so I think this is ready.

**Pull request merged**